Silent the warning: comparison between signed and unsigned integer expressions

In Travis log with commit 0f72498:

```
ht.c: In function ‘compute_size’:
ht.c:55:13: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  return idx != UT32_MAX ? ht_primes_sizes[idx] : (sz | 1);
             ^
ht.c:55:48: warning: signed and unsigned type in conditional expression [-Wsign-compare]
  return idx != UT32_MAX ? ht_primes_sizes[idx] : (sz | 1);
                                                ^
ht.c: In function ‘ht_new_size’:
ht.c:128:11: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  while (i < S_ARRAY_SIZE (ht_primes_sizes) &&
           ^
ht.c:129:36: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   ht_primes_sizes[i] * LOAD_FACTOR < initial_size) {
                                    ^

array.c: In function ‘sdb_array_contains’:
array.c:503:35: warning: signed and unsigned type in conditional expression [-Wsign-compare]
    size_t len = next ? next - ptr : strlen (ptr);
                                   ^
```

Comparison between signed and unsigned int could be really dangerous.
So these comparisons should be fixed properly.

Cc @ret2libc

> On 9 Oct 2018, at 08:49, lzutao <notifications@github.com> wrote:
> 
> In Travis log:
> 
> ht.c: In function ‘compute_size’:
> ht.c:55:13: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
>   return idx != UT32_MAX ? ht_primes_sizes[idx] : (sz | 1);
>              ^
> ht.c:55:48: warning: signed and unsigned type in conditional expression [-Wsign-compare]
>   return idx != UT32_MAX ? ht_primes_sizes[idx] : (sz | 1);
>                                                 ^
> ht.c: In function ‘ht_new_size’:
> ht.c:128:11: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
>   while (i < S_ARRAY_SIZE (ht_primes_sizes) &&
>            ^
> ht.c:129:36: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
>    ht_primes_sizes[i] * LOAD_FACTOR < initial_size) {
>                                     ^
> 
> array.c: In function ‘sdb_array_contains’:
> array.c:503:35: warning: signed and unsigned type in conditional expression [-Wsign-compare]
>     size_t len = next ? next - ptr : strlen (ptr);
>                                    ^
> Comparison between signed and unsigned int could be really dangerous.
> So these comparisons should be fixed properly.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

--- a/src/array.c
+++ b/src/array.c
@@ -500,7 +500,7 @@ SDB_API bool sdb_array_contains(Sdb *s, const char *key, const char *val, ut32 *
 		size_t vlen = strlen (val);
 		while (1) {
 			next = strchr (ptr, SDB_RS);
-			size_t len = next ? next - ptr : strlen (ptr);
+			size_t len = next ? (size_t)(next - ptr) : strlen (ptr);
 			if (len == vlen && !memcmp (ptr, val, len)) {
 				return true;
 			}

--- a/src/ht.c
+++ b/src/ht.c
@@ -7,7 +7,7 @@
 #define S_ARRAY_SIZE(x) (sizeof (x) / sizeof (x[0]))
 
 // Sizes of the ht.
-static const int ht_primes_sizes[] = {
+static const ut32 ht_primes_sizes[] = {
 	3, 7, 11, 17, 23, 29, 37, 47, 59, 71, 89, 107, 131,
 	163, 197, 239, 293, 353, 431, 521, 631, 761, 919,
 	1103, 1327, 1597, 1931, 2333, 2801, 3371, 4049, 4861,
@@ -49,7 +49,7 @@ static inline void freefn(SdbHt *ht, HtKv *kv) {
 	}
 }
 
-static inline ut32 compute_size(int idx, ut32 sz) {
+static inline ut32 compute_size(ut32 idx, ut32 sz) {
 	// when possible, use the precomputed prime numbers which help with
 	// collisions, otherwise, at least make the number odd with |1
 	return idx != UT32_MAX ? ht_primes_sizes[idx] : (sz | 1);
@@ -123,7 +123,7 @@ SDB_API SdbHt* ht_new(DupValue valdup, HtKvFreeFunc pair_free, CalcSize calcsize
 }
 
 SDB_API SdbHt* ht_new_size(ut32 initial_size, DupValue valdup, HtKvFreeFunc pair_free, CalcSize calcsizeV) {
-	int i = 0;
+	ut32 i = 0;
 
 	while (i < S_ARRAY_SIZE (ht_primes_sizes) &&
 		ht_primes_sizes[i] * LOAD_FACTOR < initial_size) {

